FastPriorityQueue.js

```diff
@@ -184,21 +219,21 @@ FastPriorityQueue.prototype.poll = function() {
   var ans = this.array[0];
   if (this.size > 1) {
     this.array[0] = this.array[--this.size];
-    this._percolateDown(0 | 0);
+    this._percolateDown(0);
   } else {
     this.size -= 1;
   }
   return ans;
 };
 
 // This function adds the provided value to the heap, while removing
-//  and returning the peek value (like poll). The size of the priority
+// and returning the smallest value (like poll) . The size of the queue
```

Comment:
Again, it is *a* smallest value since we don't know that there is just one smallest.